signac/contrib/migrations.py

```diff
@@ -0,0 +1,100 @@
+# Copyright (c) 2019 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+import sys
+from abc import abstractmethod
+from packaging import version
+
+from ..common.config import get_config
+from ..version import __version__, SCHEMA_VERSION
+
+
+FN_MIGRATION_LOCKFILE = '.SIGNAC_PROJECT_MIGRATION_LOCK'
+
+
+class Migration:
+
+    @abstractmethod
+    def apply(self, project):
+        pass
+
+    @abstractmethod
+    def rollback(self, project):
+        pass
+
+
+class Migrate0To1(Migration):
+
+    def apply(self, project):
+        pass  # nothing to do
+
+    def rollback(self, project):
+        pass  # nothing to do
+
+
+MIGRATIONS = {
+    ('0', '1'):    Migrate0To1(),
+}
+
+
+def _reload_project_config(project):
+    project_reloaded = project.get_project(
+        root=project._rd, search=False, _ignore_schema_version=True)
+    project._config = project_reloaded._config
+
+
+def _update_project_config(project, **kwargs):
+    "Update the project configuration, for internal use only."
```

Comment:
Isn't "internal use only" implied by the fact that it's a private method?

Reply:
Yes.